File: Data Structures & Algorithms/evaluate-reverse-polish-notation/submission-0.cpp
class Solution {
private:
    bool isOperator(std::string s) {
        return s=="+" || s=="-" || s == "/" || s=="*";
        }
public:
    

    int evalRPN(vector<string>& tokens) {
        std::stack<int> st;

        
        for (int i = 0; i< tokens.size(); i++) {
                if (isOperator(tokens[i])) {
                    int b = st.top(); st.pop();
                    int a = st.top(); st.pop();
                    if (tokens[i] == "+")
                            st.push(a + b);
                        else if (tokens[i] == "-")
                            st.push(a - b);
                        else if (tokens[i] == "/")
                                st.push(a/b);
                            else if (tokens[i] == "*")
                                    st.push(a*b);
                }
                else {
                    int val = std::stoi(tokens[i]);
                        st.push(val);
                }
        }
    return  st.top();
    }           
};

File: Data Structures & Algorithms/validate-parentheses/submission-0.py
class Solution:
    def isValid(self, s: str) -> bool:
        opening = [ '(', '[', '{' ]
        closing = [ ')', ']', '}' ]
        stack = []
        for i in s:
            if i in opening:
                stack.append(i)
            elif i in closing:
                if len(stack) == 0:
                    return False
                elif opening.index(stack[-1]) == closing.index(i):
                    stack.pop()
                else:
                    return False
        
        if len(stack) == 0:
            return True
        else:
            return  False
            